src/winerror.cr

```diff
@@ -0,0 +1,2104 @@
+class WinError < Exception
```

Comment:
I deeply disagree with creating a `WinError` class. `Errno` is platform-specific, and it needs to go. Exception hierarchy should never depend on the platform you're running on.